core/src/main/scala/cats/Inject.scala

```diff
@@ -0,0 +1,54 @@
+package cats
+
+/**
+ * Inject is a type class to inject a value of type `A` into a
+ * coproduct of types `B` when `A :≺: B`. The coproduct `B` is
+ * modelled with `Either`.
```

Comment:
It's true that it is the original incentive for introducing this class, but now that we made a type class, this is just one specific usage/instances of `Inject` and `InjectK` right? 
We could have an `Inject` between `A` and `List[A]` or `Int` and `Double`, right?